src/Model.ts

```diff
@@ -323,33 +327,55 @@ export abstract class Model
         return relationNames;
     }
 
-    /**
-     * @returns {string} e.g. 'http://www.foo.com/bar/'
-     */
-    public abstract getJsonApiBaseUrl(): string;
+    public static getJsonApiBaseUrl(): string {
+      if (this.jsonApiBaseUrl === undefined) {
+        throw new Error(`Expected ${this.name} to have property expect jsonApiBaseUrl defined`)
+      }
+      return this.jsonApiBaseUrl;
+    }
+
+    public static getJsonApiType(): string {
+      if (this.jsonApiType === undefined) {
+        throw new Error(`Expected ${this.name} to have property expect jsonApiType defined`)
+      }
+      return this.jsonApiType;
+    }
+
+    private static getEndpoint(): string {
+        return this.endpoint ?? `/${this.getJsonApiType()}`;
+    }
+
+    public static getJsonApiUrl(): string {
+      return `${this.getJsonApiBaseUrl()}${this.getEndpoint()}`
+    }
 
-    /**
-     * Allows you to get the current HTTP client (AxiosHttpClient by default), e.g. to alter its configuration.
-     * @returns {HttpClient}
-     */
     public static getHttpClient(): HttpClient
     {
-        return this.httpClient;
+      if (!this.httpClient) {
+        this.httpClient = new AxiosHttpClient();
+      }
+      return this.httpClient
     }
 
     /**
-     * Allows you to use any HTTP client library, as long as you write a wrapper for it that implements the interfaces
-     * HttpClient, HttpClientPromise and HttpClientResponse.
-     * @param httpClient
+     * @deprecated Use the static method with the same name instead
      */
-    public static setHttpClient(httpClient: HttpClient)
-    {
-        this.httpClient = httpClient;
+    public getJsonApiType(): string {
+      return (this as Model).constructor.getJsonApiType()
     }
 
-    public getJsonApiType(): string
-    {
-        return this.jsonApiType;
+    /**
+     * @deprecated Use the static method with the same name instead
+     */
+    public getJsonApiBaseUrl(): string {
+      return (this as Model).constructor.getJsonApiBaseUrl()
+    }
```

Comment:
I'm not sure I would deprecate this method since `type` is part of a JSON:API resource.

Reply:
(Assuming you mean the comment on `getJsonApiType()`. In JSON:API this property is available on every instance of every resource. On the JSON:API level this is the way to communicate the type of an instance.

However on the "object model" level to which Coloquent translates resources, also the way of communicating the type is translated, i.e. by actually creating an object in the right type, so the type of the model can be obtained by  `.constructor` (or by doing an `instanceof` check). Since the object model is supposed to be an abstraction from the JSON:API/resource layer, this is the type people should use if they want to work with the types of the models. However if one still needs to get not the type of the object itself, but its corresponding JSON:API type, then one can ask the object type rather than the object, as it will vary by type and not by instance, so e.g. do `(new Teacher()).constructor.jsonApiType` or `Teacher.jsonApiType`. As of now `Teacher.getJsonApiType()` is also a public method and a way to do this, but is longer syntax and exists only so that the library internally does a runtime check on the input, so perhaps it should be made private, to simplify the API surface for the user. I'll leave that for another time.

Reply:
My bad haha, I meant `getJsonApiType()` like you said. Ok, I understand, you want to keep the abstraction from JSON:API in the model layer. 👍